scripts/generate-script-readmes.ts

```diff
@@ -0,0 +1,241 @@
+import { promises as fs } from 'fs';
+import yaml from 'js-yaml';
+
+const maybeIntegrations = await fs.readdir('integrations', { withFileTypes: true });
+const integrations = maybeIntegrations.filter((dirent) => dirent.isDirectory()).map((dirent) => dirent.name);
+
+for (const integration of integrations) {
+    const yamlConfig = yaml.load(await fs.readFile(`integrations/${integration}/nango.yaml`, 'utf8')) as any;
+    const config = yamlConfig.integrations[integration] || yamlConfig.integrations['${PWD}'];
+    const models = yamlConfig.models || {};
+
+    if (config.syncs) {
+        (Object.entries(config.syncs) as [string, any]).map(async ([key, sync]) => {
+            let sections = await readSections(`integrations/${integration}/syncs/${key}.md`);
+            try {
+                sections = updateReadme(sections, key, `${integration}/syncs/${key}`, 'Sync', sync, models);
+                await fs.writeFile(`integrations/${integration}/syncs/${key}.md`, readme(sections));
+            } catch (e) {
+                console.error(`Error updating readme for ${integration} sync ${key}: ${e.message}`);
+            }
+        });
+    }
+
+    if (config.actions) {
+        (Object.entries(config.actions) as [string, any]).map(async ([key, action]) => {
+            let sections = await readSections(`integrations/${integration}/actions/${key}.md`);
+
+            try {
+                sections = updateReadme(sections, key, `${integration}/actions/${key}`, 'Action', action, models);
+                await fs.writeFile(`integrations/${integration}/actions/${key}.md`, readme(sections));
+            } catch (e) {
+                console.error(`Error updating readme for ${integration} action ${key}: ${e.message}`);
+            }
+        });
+    }
+}
+
+type MarkdownSections = { [key: string]: string[] };
+
+// function to parse markdown into sections
+async function readSections(filename: string): Promise<MarkdownSections> {
+    let markdown;
+    try {
+        markdown = await fs.readFile(filename, 'utf8');
+    } catch {
+        markdown = '';
+    }
+
+    const lines = markdown.split('\n');
+    const sections: MarkdownSections = {};
+    let currentSection = '';
+
+    for (const line of lines) {
+        const sectionMatch = line.trim().startsWith('#');
+        if (sectionMatch) {
+            currentSection = line.trim();
+            sections[currentSection] = [];
+        } else if (currentSection) {
+            sections[currentSection].push(line);
+        }
+    }
+
+    return sections;
+}
+
+// update sections to include latest content
+function updateReadme(
+    sections: MarkdownSections,
+    scriptName: string,
+    scriptPath: string,
+    endpointType: string,
+    scriptConfig: any,
+    models: any
+): MarkdownSections {
```

Comment:
Why not splitting the file in two (with a well know pattern) and just always update everything. Seems a bit simpler, especially if you add/rename/remove section

Reply:
@bodinsamuel can you say more?

Reply:
See a381ed0. Is that more like what you're thinking?

Reply:
seems much easier to maintain no? 👍🏻

Reply:
Yeah, previously the issue called for it to maintain these section titles while allowing human edits in between, but I think where this ended up we're set to just treat it as two big blocks.